tests/signal_manager_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+. ./src/signal_manager.sh --source-only
+. ./tests/utils --source-only
+
+function suite() {
+  suite_addTest "default_handler_Test"
+  suite_addTest "non_signal_Test"
+  suite_addTest "new_signal_Test"
+  suite_addTest "signal_reset_Test"
+  suite_addTest "add_two_signals_Test"
+}
+
+function default_handler_Test() {
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'$SEPARATOR$'\n'Exiting...
+  local output
+
+  cd "$SHUNIT_TMPDIR"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
+    wait $!
+  } > out.tmp &
+
+  $(echo :)
+  kill -s SIGTERM $!
+  assertEquals "$expected" "$(cat out.tmp)"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
+    wait $!
+  } > out.tmp &
+
+  $(echo :)
```

Comment:
Could somebody help me understand this behaviour? It seems taht for some reason the output of the grouped command (lines 31-35) only gets redirected to `out.tmp` if a subshell is created between that and the `kill` command.

Reply:
My current working hypothesis is that there is a synchronization issue between the setting of the trap (l. 34) and the sending of the signal (l. 40). Calling a subshell between those only seemed to work (actually, it didn't work every time) because it favors the setting of the trap in the race for execution.

Reply:
Tbh, I did not get why we need this `$(echo :)`, is it some bash-magic?

Anyway, when I was working with `kw pomodoro` I noticed that the subprocess takes a while to start (make sense, since starting a new process is a heavyweight task). In this sense, this part:

```
   {
      signal_manager default_interrupt_handler SIGTERM SIGINT
      sleep infinity &
      wait $!
    } > out.tmp &
```

might take a while to start. You can check this hypothesis by adding a `sleep 1s` right after the above code.

Reply:
Indeed, using `sleep 1` instead of `$(echo :)` works. In the last update of this PR I came up with a system to make sure the subprocess was already created before sending it the signal, using more signals. I think I was able to remove all the race conditions without having to use `sleep`. 

By the way, `$(echo :)` makes no sense at all. I thought of using the builtin `:` but instead passed it as an argument to echo.